tests/src/test/scala/system/basic/WskSdkTests.scala

```diff
@@ -82,6 +84,76 @@ class WskSdkTests extends TestHelpers with WskTestHelpers {
     }
   }
 
+  it should "download docker sdk when blackbox.tar.gz exists, and docker name should be quoted in error." in {
+    val fileName = "blackbox.tar.gz"
+    val dir = File.createTempFile("wskinstall", ".tmp")
+    dir.delete()
+    dir.mkdir() should be(true)
+    val file = new File(dir, fileName)
+    file.createNewFile should be (true)
+    try {
+      wsk.cli(wskprops.overrides ++ Seq("sdk", "install", "docker"),
+        workingDir = dir,
+        expectedExitCode = 1).stderr should include(
+        s"""The file '${fileName}' already exists.  Delete it and retry.""")
+    } finally {
+      file.delete()
+      FileUtils.deleteDirectory(dir)
+    }
+  }
+
+  it should "download ios sdk when OpenWhiskIOSStarterApp.zip exists, and ios name should be quoted in error." in {
+    val fileName = "OpenWhiskIOSStarterApp.zip"
+    val dir = File.createTempFile("wskinstall", ".tmp")
+    dir.delete()
+    dir.mkdir() should be(true)
+    val file = new File(dir, fileName)
+    file.createNewFile should be (true)
+    try {
+      wsk.cli(wskprops.overrides ++ Seq("sdk", "install", "ios"),
+        workingDir = dir,
+        expectedExitCode = 1).stderr should include(
+        s"""The file '${fileName}' already exists.  Delete it and retry.""")
+    } finally {
+      file.delete()
+      FileUtils.deleteDirectory(dir)
+    }
+  }
+
+  it should "download ios sdk, check filename is quoted in error when create fails." in {
```

Comment:
Can we expect the same directory permission issue with the Docker SDK?

```
$ mkdir wskinstall
$ chmod 555 wskinstall
$ cd wskinstall
$ wsk sdk install docker
error: The docker SDK installation failed: Error creating SDK file 'blackbox.tar.gz': open blackbox.tar.gz: permission denied
```

Could test both with `Seq`.

Reply:
I'll have to double check this one but last i checked i think there was no way to inject the bad permissions between the download and extraction.